### PR TITLE
TW-56891 Support Docker Wrapper, upgrade gradle

### PR DESCRIPTION
**Pull request by vmayushan**

### CODE REVIEW

File: MAINTAINERSHIP.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    ```shell
+    ./tag-release-and-push.sh
+    ```
+
+It will tag the current `HEAD` with latest version from the changelog, and push it to the origin remote.
+
+The new version of the plugin will be published to [marketplace][marketplace.plugin-page] automatically.
+
+[semver]: https://semver.org/spec/v2.0.0.html
+[marketplace.plugin-page]: https://plugins.jetbrains.com/plugin/11453-unity-support
```
Comment (lesley29) [inline MAINTAINERSHIP.md]: fix

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+cucumber = { module = "io.cucumber:cucumber-java", version = "4.2.0" }
+commons-cli = { module = "commons-cli:commons-cli", version = "1.4" }
+constraint-transitive-icu4j = { module = "com.ibm.icu:icu4j", version = { require = "3.4.4" } }
+teamcity-service-messages = { module = "org.jetbrains.teamcity:serviceMessages", version.ref = "teamcity" }
+teamcity-internal-agent = { module = "org.jetbrains.teamcity.internal:agent", version.ref = "teamcity" }
+teamcity-internal-server = { module = "org.jetbrains.teamcity.internal:server", version.ref = "teamcity" }
+teamcity-common-api = { module = "org.jetbrains.teamcity:common-api", version.ref = "teamcity" }
+
+
+
```
Comment (lesley29) [inline gradle/libs.versions.toml]: nit: too many spaces

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
```
Comment (lesley29) [inline gradle/wrapper/gradle-wrapper.properties:3]: Hasn't anything else changed as well with this update?

File: plugin-bazel-common/build.gradle [hunk 4]
```diff
@@ -1,34 +1,31 @@
...
+    testImplementation libs.jmock
+    testImplementation libs.hamcrest
+    testImplementation libs.mockk
 }
 
-compileTestKotlin {
-    kotlinOptions.jvmTarget = "1.8"
+jar {
+    // don't include jar version into the name
+    archiveFileName = "plugin-bazel-common.jar"
```
Comment (lesley29) [inline plugin-bazel-common/build.gradle]: Could you please clarify why?

File: plugin-bazel-server/build.gradle [hunk 5]
```diff
@@ -1,21 +1,42 @@
...
+            name = "bazel"
+            displayName = "Bazel build support"
+            description = "Provides build facilities for bazel projects"
+            downloadUrl = "https://github.com/JetBrains/teamcity-bazel-plugin"
+            version = project.version.toString()
+            vendorName = "JetBrains"
+            vendorUrl = "https://www.jetbrains.com/"
+            useSeparateClassloader = true
+            allowRuntimeReload = true
+            minimumBuild = "40000"
```
Comment (lesley29) [inline plugin-bazel-server/build.gradle:25]: seems we're missing a `node-responsibilities-aware`

File: plugin-bazel-server/build.gradle [hunk 6]
```diff
@@ -24,19 +45,24 @@ dependencies {
...
+        implementation(libs.constraint.transitive.icu4j) {
+            because 'previous versions have faulty jar files which cause problems during incremental compilation (which is enabled by default since Kotlin 1.8.20)'
+        }
+    }
 }
 
-serverPlugin.version = null
+jar {
+    // don't include jar version into the name
+    archiveFileName = "plugin-bazel-server.jar"
```
Comment (lesley29) [inline plugin-bazel-server/build.gradle]: once again, we'll need to check the necessity of this change

**Review by lesley29 — APPROVED** [review]
lgtm

File: MAINTAINERSHIP.md (hunk 1 above)
Comment (vmayushan) [reply MAINTAINERSHIP.md]: fixed

File: gradle/libs.versions.toml (hunk 2 above)
Comment (vmayushan) [reply gradle/libs.versions.toml]: done

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (vmayushan) [reply gradle/wrapper/gradle-wrapper.properties:3]: i had to run `./gradlew wrapper --gradle-version 8.9` twice, and binaries were updated, found this in the doc:
```
If you want all the wrapper files to be completely up-to-date, you will need to run the wrapper task a second time.
```

https://docs.gradle.org/current/userguide/gradle_wrapper.html

File: plugin-bazel-server/build.gradle (hunk 5 above)
Comment (vmayushan) [reply plugin-bazel-server/build.gradle:25]: done

**Pull request merged**